Implement L2 Token Contracts for Optimism

GoodluckH opened the pull request:
A new deployment script is added to support L2 token creation on Optimism networks. 

Added several fields to `.env.example` for L2 token deployment related secrets/variables.

Run `npm run deployL2TokenOptimismKovan` to test out L2 token contract creation.

Also added a note to `README` to address an build cache related issue when verifying contracts.

BrassLion commented on .env.example:
```diff
@@ -9,3 +9,7 @@ DEFENDER_TEAM_API_SECRET_KEY=<SECRET_KEY>
+# L2 token deployment required secrets
+L1_TOKEN_ADDRESS=
+L2_TOKEN_NAME=Arrow Air
+L2_TOKEN_SYMBOL=ARROW
```
Let's hardcode these into the deployment script. There's no way they'd change compared to L1.

BrassLion commented on README.md:
```diff
@@ -58,6 +58,9 @@ npx hardhat deploy --network rinkeby --tags Deployment
 Deploying to mainnet is possible by subsituting all `--network rinkeby` arguments with `--network mainnet`.
 
+
+*Note: You may need to run `npx hardhat clean` to clear all the build caches and artifacts if you encounter "no such file or directory" issues when verifying contracts (also see [here](https://github.com/NomicFoundation/hardhat/issues/1117)).*
```
Did you find this solution after a pretty quick google? If so, we can probably remove this from the codebase, or move it to an issue if we really want to keep track of it.

BrassLion commented on deploy/deployArrowTokenL2.js:
```diff
@@ -0,0 +1,41 @@
+    "0x4200000000000000000000000000000000000012",
+    L2StandardTokenFactoryArtifact.abi,
+    signer
+  );
```
This can only be run on Optimism, so a check that we are running on the right chain would be useful here.

BrassLion reviewed (approved):
LGTM

BrassLion commented:
It would be great if we could get the tests passing.

GoodluckH committed:
Implement L2 Token Contracts for Optimism

A new deployment script is added to support L2 token creation on
Optimism networks.

Added several fields to .env.example for L2 token deployment related
secrets/variables.

Run npm run deployL2TokenOptimismKovan to test out L2 token contract
creation.

Also added a note to README to address an build cache related issue when
verifying contracts.

GoodluckH commented:
Thanks. Resolved all the conflicts in `package-lock.json` that's causing the test CLI to fail.
